verticapy/toolbox.py

```diff
@@ -1610,3 +1610,300 @@ def __round__(self, x):
 
     def category(self):
         return self.category_
+
+#
+#
+# Tools to merge similar names/categories together.
+#
+# ---#
+def erase_prefix_in_name(name: str, prefix: list = []):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of prefixes from the input name and returns it.
+When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+prefix: list, optional
+    List of prefixes.
+
+Returns
+-------
+name
+    The name without the prefixes.
+    """
+    name_tmp = name
+    for p in prefix:
+        n = len(p)
+        if p in name_tmp and name_tmp[:n] == p:
+            name_tmp = name_tmp[n:]
+            break
+    return name_tmp
+
+# ---#
+def erase_suffix_in_name(name: str, suffix: list = []):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of suffixes from the input name and returns it.
+When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+suffix: list, optional
+    List of suffixes.
+
+Returns
+-------
+name
+    The name without the suffixes.
+    """
+    name_tmp = name
+    for s in suffix:
+        n = len(s)
+        if s in name_tmp and name_tmp[-n:] == s:
+            name_tmp = name_tmp[:-n]
+            break
+    return name_tmp
+
+# ---#
+def erase_word_in_name(name: str, word: list = []):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of words from the input name and returns it.
+When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+word: list, optional
+    List of words.
+
+Returns
+-------
+name
+    The name without the input words.
+    """
+    for w in word:
+        if w in name:
+            return name.replace(w, "")
+            break
+    return name
+
+# ---#
+def erase_in_name(
+    name: str, 
+    suffix: list = [], 
+    prefix: list = [],
+    word: list = [],
+    order: list = ["p", "s", "w"]
+):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of suffixes and prefixes from the input name and 
+returns it. When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+suffix: list, optional
+    List of suffixes.
```

Comment:
@oualib  Why do we want only 1 suffix to be removed and rest to be ignored?

Reply:
I wanted to only test one. these functions are used to only simplify one future function.